src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,103 @@
+import { InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderRequestBody } from '../../../order';
+import { PaymentMethodFailedError } from '../../errors';
+import Payment from '../../payment';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { GooglePayPaymentStrategy, PaymentMethodData } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+const TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+const LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+export default class CheckoutcomGooglePayPaymentStrategy extends GooglePayPaymentStrategy {
```

Comment:
I read your comment regarding why you can't use the common `GooglePayPaymentStrategy` and instead have to use inheritance. If what you're saying is true, the primary hurdle is that `parseResponse` is not async, then I'd say making it an async method (returning `Promise<TokenizePayload>` instead of `TokenizePayload`) should be a trivial change. You just then to update the interface and code to use `async/await` or `catch/then`. It's better to do that than introducing a snowflake that goes against the design of other Google Pay payment strategies.

Reply:
Is there a reason why we still have `CheckoutcomGooglePayPaymentStrategy`? I can see for customer strategies, we're using `GooglePayCustomerStrategy`. Is there a reason why we can't use `GooglePayPaymentStrategy` instead rather than using an inherited class?